Tutorials/intro-to-octave/set_octave.py

```diff
@@ -161,24 +160,25 @@ def octave_settings(qmm, qm, prog, octave_config, external_clock=False, calibrat
     ###################################
     # setting down-converters modules #
     ###################################
-    for i in range(len(octave_elements)):
-        # This assumes that: FR1in measures RF1's output, FR2in measures RF2's output
-        if (
-            qm.octave._get_element_octave_output_port(octave_elements[i])[1] == 1
-            and "outputs" in config["elements"][octave_elements[i]].keys()
-        ):
-            qm.octave.set_qua_element_octave_rf_in_port(octave_elements[i], "octave1", 1)
-            qm.octave.set_downconversion(octave_elements[i])
-            qm.octave.set_downconversion(octave_elements[i], lo_source=RFInputLOSource.Internal)  # Can change to Dmd1LO
-        if (
-            qm.octave._get_element_octave_output_port(octave_elements[i])[1] == 2
-            and "outputs" in config["elements"][octave_elements[i]].keys()
-        ):
-            qm.octave.set_qua_element_octave_rf_in_port(octave_elements[i], "octave1", 2)
-            qm.octave.set_downconversion(octave_elements[i])
-            qm.octave.set_downconversion(
-                octave_elements[i], lo_source=RFInputLOSource.Dmd2LO
-            )  # Don't forget to connect external LO to Dmd2LO or Synth2 from back panel
+    for elements in octave_elements:
+        element_i = qm.elements[elements]
+        if isinstance(element_i, ElementWithOctave):
+            # This assumes that: FR1in measures RF1's output (which is connected to Analog output 1 and 2), FR2in measures RF2's output (which is connected to Analog output 3 and 4)
+            if (element_i.q_port == 1 or element_i.q_port == 2) and 'outputs' in \
+                    config['elements'][octave_elements[i]].keys():
+                qm.octave.set_qua_element_octave_rf_in_port(octave_elements[i], "octave1", 1)
+                qm.octave.set_downconversion(octave_elements[i])
+                qm.octave.set_downconversion(octave_elements[i],
+                                             lo_source=RFInputLOSource.Internal)  # Can change to Dmd1LO
+                qm.octave._set_downconversion_if_mode(octave_elements[i], if_mode_i=IFMode.direct,
```

Comment:
This is still a protected API (with "_"), is it normal?

Reply:
This is still working so I guess it's fine... @yuvalqm ?

Reply:
yes this is internal. you should use the parameters at qm.octave.set_downconversion for the if I/Q modes for example: 
`        qm.octave.set_downconversion(elements[0], lo_source=RFInputLOSource.Internal,if_mode_i= IFMode.direct, if_mode_q=IFMode.direct)
`

Reply:
Changed it. Thanks!